### ISSUE TITLE
Many compile errors in window7 64

### ISSUE THREAD

**Issue by wguo68** [JoeyDeVries/LearnOpenGL#25]
>basic_lighting_specular.obj : error LNK2019: unresolved external symbol glewInit referenced in function main
1>basic_lighting_specular.obj : error LNK2019: unresolved external symbol glfwInit referenced in function main
1>basic_lighting_specular.obj : error LNK2019: unresolved external symbol glfwTerminate referenced in function main
1>basic_lighting_specular.obj : error LNK2019: unresolved external symbol glfwWindowHint referenced in function main
1>basic_lighting_specular.obj : error LNK2019: unresolved external symbol glfwCreateWindow referenced in function main
1>basic_lighting_specular.obj : error LNK2019: unresolved external symbol glfwWindowShouldClose referenced in function main
1>basic_lighting_specular.obj : error LNK2019: unresolved external symbol glfwSetWindowShouldClose referenced in function "void __cdecl key_callback(struct GLFWwindow *,int,int,int,int)" (?key_callback@@YAXPEAUGLFWwindow@@HHHH@Z)
1>basic_lighting_specular.obj : error LNK2019: unresolved external symbol glfwPollEvents referenced in function main
1>basic_lighting_specular.obj : error LNK2019: unresolved external symbol glfwSetInputMode referenced in function main

**Comment by JoeyDeVries** [comment]
It couldn't link properly with GLFW from the looks of it. Try building GLFW yourself (or get one of their pre-compiled binaries for your system if available). Link against that library and it should work

**Comment by wguo68** [comment]
Yes,when I build glfw myself.

**Comment by gameengineer** [comment]
That's a hassle though.  You include the lib binaries so you should fix your build problem not us. Previous releases built fine.

**Comment by JoeyDeVries** [comment]
It's been scratching my head for a while to indirectly link to GLFW's source in the CMAKE file and re-compile the library whenever the project is compiled for the first time, but I'm not an expert at CMAKE so I'll have to contribute quite some time to figure this out; time I don't have at the moment. I've put it on the list and plan to work on it soon.

**Comment by gameengineer** [comment]
I guess I hadn't thought of that.  You have to account for a host of different builds with CMake. My apologies for coming across too abrupt with my comments to you.  Your code and explanations have helped me immensely.

**Comment by JoeyDeVries** [comment]
Not at all, they're not critical concerns, but valid concerns nonetheless. Once I'm a bit better at CMAKE I'll probably give it a shot again ;)

**Comment by iiRecord** [comment]
well, the glfw3.lib is now compiled with vs2015 . the lib not work in vs2012, i have to recompile glfw in vs2012. it works too.

**Comment by aboutqx** [comment]
Hi,i'm using cmake to generate files for vs2015 on win7 64,and still get errors below.I've download glfw64.zip,how shuold i replace the glfw?It got 3 files,glfw3.dll,glfw3.lib,glfw3dll.lib.

**Comment by zoozooll** [comment]
cmake select **WIN32** . OK

**Comment by aboutqx** [comment]
cmake option win32 works， don‘t need rebuild glfw from source.
